### PR TITLE
Add offer and accept flow for numeric events

### PR DESCRIPTION
**Pull request by benthecarman**
Needs testing and probably some code cleanup

### CODE REVIEW

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/CreateDLCOfferDialog.scala [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+import scalafx.scene.layout._
+import scalafx.stage.Window
+
+import java.time.ZoneOffset
+import scala.collection._
+import scala.util.{Failure, Success, Try}
+
+object CreateDLCOfferDialog {
+
+  def showAndWait(parentWindow: Window): Option[CreateDLCOffer] = {
```
Comment (Christewart) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/CreateDLCOfferDialog.scala:26]: This method is 400 lines. We need to break this up and move all the helper methods out of it.

Comment (benthecarman) [reply app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/CreateDLCOfferDialog.scala:26]: Did some simplifications, however a lot of this requires the elements in the gui

**Comment by Christewart** [comment]
Needs rebase

**Review by Christewart — APPROVED** [review]

**Comment by Christewart** [comment]
Needs rebase and CI is failing 

https://github.com/bitcoin-s/bitcoin-s/pull/3036/checks?check_run_id=2567959443#step:5:3693

**Review by Christewart — APPROVED** [review]

**Pull request merged**